### DISCUSSION TITLE
How can we set a custom cursor in MRTK 3 for far ray

### DISCUSSION THREAD

**Discussion by amea32179** [microsoft/MixedRealityToolkit-Unity#11688] [General]
How can we set a custom cursor in MRTK 3 for far ray

**Comment by marlenaklein-msft** [comment]
Hi @amea32179, 

On the MRTK XR Rig > CameraOffset > MRTK RightHand Controller > Far Ray there is a component MRTK Ray Reticle Visual. You will want to set the **Base Reticle** to the cursor game object you want, and **Reticle Root** to a transform that controls the position of that game object. 

(Also, if you want a custom cursor only when hovering a specific Interactable, under Base Interactable Settings > Base XRI Settings you can set a custom reticle for that Interactable).

**Comment by amea32179** [comment]
Hi @marlenaklein-msft,
            I just checked with the above mentioned way. But, we could not set a custom cursor at runtime since **Base Reticle and Reticle Root are not accessible**. Is there a way to set custom cursor at runtime?

**Comment by marlenaklein-msft** [comment]
Hi @amea32179,

You're correct, currently these cannot be set at runtime. Could you open a feature request for this? We will look into if these should be exposed as public properties for a future release. Thanks!

**Comment by amea32179** [comment]
Thanks @marlenaklein-msft for the quick reply. Have create the feature request - https://github.com/microsoft/MixedRealityToolkit-Unity/issues/11702

**Comment by AMollis** [comment]
@amea32179 can you please describe your scenario a bit more?   In what scenario are you trying to change the "cursor" (or "reticle")?  

If you are just wanting to change the reticle when hovering over an interactable, have you tried using the "custom reticle" property on the interactable itself.  This will change the cursor when you hover over the interactable.

![image](https://github.com/microsoft/MixedRealityToolkit-Unity/assets/36461279/723a13a0-6313-4e41-8a66-2d2abb7ef28e)

If you want to change the cursor, regardless of what interactable is being hovered, I recommend finding the `BaseReticleVisual` on the interactor, and call the `BaseReticleVisual.AttachCustomReticle`.  This will hide the default reticle, and show your custom one.  Note, as soon as you hover over an interactable with a custom reticle set, this will be cleared.

If you want to change the cursor regardless of what interactable is being hovered and without calling `BaseReticleVisual.AttachCustomReticle`, you'll need create a variant of the `MRTK LeftHand Controller` and change the Far Ray's reticle to your own game object.  Your game object could then manage its own visual as it sees fit.

![image](https://github.com/microsoft/MixedRealityToolkit-Unity/assets/36461279/37f4d244-891c-4691-bc50-9bf90d5df370)

At this time, we don't believe it necessary to expose the underlying reticle game objects.
